File: prdeploy-api/src/PrDeploy.Api.Business/Models/Settings/RepoSettings.cs
namespace PrDeploy.Api.Business.Models.Settings;
public class RepoSettings
{
    public string? Owner { get; set; }
    public string? Repo { get; set; }
    public string? DeployWorkflow { get; set; }
    public List<EnvironmentSettings>? Environments { get; set; } = new ();
    public List<ServiceSettings>? Services { get; set; } = new();
    public string? DefaultEnvironment { get; set; }
    public string? ReleaseEnvironment { get; set; }
    public string? SettingsBranch { get; set; }
    public bool? AddJiraIssues { get; set; }
    public BuildsSettings? Builds { get; set; }
    public SlackSettings? Slack { get; set; }
    public EmailAliases? EmailAliases { get; set; }
    public string? DeployManagerSiteUrl { get; set; }
    public BadgeSettings? Badge { get; set; }
}

File: prdeploy-webhooks/src/models/settings/repo-settings.ts
import { Lifecycle, scoped } from 'tsyringe';
import { EmailAliases } from '../email-aliases';
import { EnvironmentSettings } from './environments-settings';
import { SlackSettings } from './slack-settings';
import { BuildsSettings } from './builds-settings';
import { BadgeSettings } from './badge-settings';
import { ServiceSettings } from './service-settings';

@scoped(Lifecycle.ContainerScoped)
export class RepoSettings {
  owner: string;
  repo: string;
  deployWorkflow: string;
  syncWorkflow?: string;
  environments: EnvironmentSettings[];
  services: ServiceSettings[];
  defaultEnvironment: string;
  releaseEnvironment: string;
  defaultBranch: string;
  settingsBranch: string;
  addJiraIssues: boolean;
  builds: BuildsSettings;
  slack: SlackSettings;
  emailAliases: EmailAliases;
  deployManagerSiteUrl: string;
  badge: BadgeSettings;
}
